lib/Twig/RuntimeLoader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Creates the runtime implementation of a Twig element (filter/function/test).
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class Twig_RuntimeLoader implements Twig_RuntimeLoaderInterface
+{
+    private $map;
+
+    /**
+     * @param array $map An array of format [class => factory callable]
+     */
+    public function __construct(array $map = array())
+    {
+        $this->map = $map;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load($class)
+    {
+        if (isset($this->map[$class])) {
+            $runtime = $this->map[$class];
+
+            return $runtime();
+        }
+
+        return null;
```

Comment:
IIRC the `return` statement can then simply be omitted to match the Twig code style rules.